src/configs/components.js

```diff
@@ -0,0 +1,40 @@
+const components = {
+    Button: 'button',
+    IconButton: 'button',
+    Link: 'a',
+    Spinner: 'svg',
+    Octicon: 'svg',
+    ToggleSwitch: 'button',
+    Radio: 'input',
+    TextInput: {
+        Action: 'button',
+        self: 'input',
+    }, 
+    Select: {
+        Option: 'option', 
+        self: 'select',
+     },
+    TabNav: {
+        Link: 'a',
+        self: 'nav',
+    },
+}
+
+const flattenComponents = (componentObj) => {
```

Comment:
This flattens the object passed by 1 level:

```js
const components = {
    Select: {
        Option: 'option', 
        self: 'select',
     },
}
```
The above example would then be flattened:

```js
const components = {
    Select: 'select',
    Select.Option: 'option', 
}
```

It also converts the child keys to have the [proper naming convention](https://primer.style/react/Select) (e.g. `Select.Option`), while `self` being the parent's element type.

Reply:
Yup, that's a good idea! Will introduce a test for it!

Reply:
Added a test for the `flattenComponents` function!